clang/include/clang/Basic/Attr.td

```diff
@@ -1275,6 +1275,20 @@ def SYCLSimdAccessorPtr : InheritableAttr {
   let Documentation = [Undocumented];
 }
 
+// Used by FE to mark SYCL kernel pointer parameters which correspond to the
+// original lambda's captured accessors. FE turns the attribute to some metadata
+// required by the FPGA back-end.
+// This attribute does not require custom semantic handling 
+// hence we set the SemaHandler field to 0.
+// The attribute is not for public consumption, and is an implicitly-created attribute
+// that has no visible spelling, hence undocumented.
+def SYCLAccessorPtr : InheritableAttr {
+  // This attribute has no spellings as it is only ever created implicitly.
+  let Spellings = [];
+  let SemaHandler = 0;
```

Comment:
You should probably add - 

`let Subjects = SubjectList<[ParmVar]>;`

Reply:
I had initially considered adding [Subjects](https://clang.llvm.org/docs/InternalsManual.html#subjects) , but looking at the Clang documentation, it appeared that it is mostly required to generate diagnostic messages for attributes used by users.
Since `SYCLAccessorPtr` is not for user consumption, I had decided to not use `Subjects` in the attribute definition.
Besides, I also see [SYCLAccessorReadonly](https://github.com/intel/llvm/blob/sycl/clang/include/clang/Basic/Attr.td#L1279) , which is also not for public consumption, drop the `Subjects` node in its definition. 
If it still needs to be added, I would like to understand why.

Reply:
You're right. We don't need it if we don't require diagnostics. I tend to add it if there is a known requirement (even if its an internal attribute) but it is not strictly required I guess. So yes, in this case you can probably skip it.